controllers/spec/common.go

```diff
@@ -1725,3 +1787,251 @@ func getFilenameOfComponentPackage(componentPackage string) string {
 	}
 	return componentPackage
 }
+
+func MakeJavaFunctionCommandDistroless(packageFile, logLevel, memory, extraDependenciesDir string) []string {
+	processCommandArgs := getProcessJavaRuntimeArgsDistroless(packageFile, logLevel, memory, extraDependenciesDir)
+	return processCommandArgs
+}
+
+func getFunctionRunnerImageDistroless(spec *v1alpha1.FunctionSpec) string {
+	runtime := &spec.Runtime
+	img := spec.Image
+	if img != "" {
+		return img
+	} else if runtime.Java != nil && runtime.Java.Jar != "" {
+		return Configs.RunnerImages.JavaDistroless
+	} else if runtime.Python != nil && runtime.Python.Py != "" {
+		log.Info("Do not support running Python distroless image yet, fall back to use old Python" +
+			" runner image")
+		return Configs.RunnerImages.Python
+	} else if runtime.Golang != nil && runtime.Golang.Go != "" {
+		log.Info("Do not support running Golang distroless image yet, fall back to use old Golang runner image")
+		return Configs.RunnerImages.Go
+	}
+	return DefaultRunnerImage
+}
+
+func getProcessJavaRuntimeArgsDistroless(packageName, logLevel, memory, extraDependenciesDir string) []string {
+	classPath := "/pulsar/instances/java-instance.jar"
+	if extraDependenciesDir != "" {
+		classPath = fmt.Sprintf("%s:%s/*", classPath, extraDependenciesDir)
+	}
+	setLogLevel := ""
+	if logLevel != "" {
+		setLogLevel = strings.Join(
+			[]string{
+				fmt.Sprintf("-Dpulsar.log.level=%s", logLevel),
+				fmt.Sprintf("-Dbk.log.level=%s", logLevel),
+			},
+			" ")
+	}
+	args := []string{
+		"java",
```

Comment:
This is where we compose the start command of the Java function runner. It only uses one config option `--config-file`. I remember there were some issue using java as the first command.